crates/core/database/src/models/admin_migrations/ops/mongodb/scripts.rs

```diff
@@ -1165,6 +1164,43 @@ pub async fn run_migrations(db: &MongoDb, revision: i32) -> i32 {
             .expect("failed to update users");
     }
 
+    if revision <= 41 {
+        info!("Running migration [revision 41 / 05-06-2025]: convert role ranks to uniform numbers.");
+
+        let mut servers = db.db()
+            .collection::<Server>("servers")
+            .find(doc! {
+                "roles": {
+                    "$exists": true,
+                    "$ne": []
+                }
+            })
+            .await
+            .unwrap()
+            .filter_map(|s| async { s.ok() })
+            .boxed();
```

Comment:
This fetch might fail, should use a clone of the Server struct containing the required information